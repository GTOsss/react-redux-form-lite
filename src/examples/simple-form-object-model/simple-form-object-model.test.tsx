import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import SimpleFormObjectModel from './simple-form-object-model';
import {reducer} from '../../index';

interface IValues {
  firstName?: string;
  lastName?: string;
  email?: string;
  sex?: string;
  favoriteColor?: string;
  employed?: string;
  notes?: string;
}

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}>
      <SimpleFormObjectModel onSubmit={onSubmit} />
    </Provider>,
  );

  return {
    reduxThunkTester,
    getFormState: () => store.getState().reduxForm.simple as IReduxFormState<IValues>,
    component,
  };
};

describe('<SimpleFormObjectModel />', () => {
  test('Register form and fields: store.', () => {
    const {getFormState} = renderComponent();
    expect(getFormState()).toMatchSnapshot();
  });

  test('Focus field: store', () => {
    const {getFormState, component} = renderComponent();
    component.find('input').at(0).simulate('focus');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Changed field: store', () => {
    const {component, getFormState} = renderComponent();
    component.find('input').at(0).simulate('focus');
    component.find('input').at(0).simulate('change', {target: {value: 't'}});
    expect(getFormState()).toMatchSnapshot();
  });

  test('Blur field: store', () => {
    const {component, getFormState} = renderComponent();
    component.find('input').at(0).simulate('focus');
    component.find('input').at(0).simulate('change', {target: {value: 't'}});
    component.find('input').at(0).simulate('blur');
    expect(getFormState()).toMatchSnapshot();
  });

  test('onSubmit', () => {
    const onSubmit = jest.fn();
    const {component} = renderComponent(onSubmit);
    component.find('form').simulate('submit');
    expect(onSubmit.mock.calls.length).toBe(1);
    expect(onSubmit.mock.calls).toMatchSnapshot();
  });

  test('Failed validation: store', () => {
    const {component, getFormState} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {value: 't'}});
    component.find('form').simulate('submit');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Submit success: store', () => {
    const {component, getFormState} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getFormState()).toMatchSnapshot();
  });
});
