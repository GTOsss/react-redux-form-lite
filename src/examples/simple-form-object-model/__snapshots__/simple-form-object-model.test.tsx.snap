// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleFormObjectModel /> Blur field: store 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": true,
    "changed": true,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": true,
        "changed": true,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": "t",
      "lastName": undefined,
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> Changed field: store 1`] = `
Object {
  "form": Object {
    "activeField": "profile.firstName",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": true,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": "t",
      "lastName": undefined,
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> Failed validation: store 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {
      "profile": Object {
        "firstName": "Field required.",
        "lastName": "Must be 2 characters or more.",
      },
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "Must be 2 characters or more.",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": undefined,
      "lastName": "t",
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> Focus field: store 1`] = `
Object {
  "form": Object {
    "activeField": "profile.firstName",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": true,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> Register form and fields: store. 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> Submit success: store 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "email": undefined,
    },
    "profile": Object {
      "firstName": "test",
      "lastName": "test",
    },
  },
}
`;

exports[`<SimpleFormObjectModel /> onSubmit 1`] = `
Array [
  Array [
    Object {
      "actions": Object {
        "arrayPush": [Function],
        "blur": [Function],
        "change": [Function],
        "changeSubmitted": [Function],
        "focus": [Function],
        "registerField": [Function],
        "registerForm": [Function],
        "removeField": [Function],
        "removeForm": [Function],
        "resetForm": [Function],
        "setInitialValues": [Function],
        "updateFormState": [Function],
        "updateValidateAndWarningMessages": [Function],
        "updateValidateMessage": [Function],
        "updateValidateMessages": [Function],
        "updateWarningMessage": [Function],
        "updateWarningMessages": [Function],
      },
      "state": Object {
        "form": Object {
          "activeField": "",
          "blurred": false,
          "changed": false,
          "errorsMap": Object {
            "profile": Object {
              "firstName": "Field required.",
              "lastName": "Field required.",
            },
          },
          "focused": false,
          "hasErrors": true,
          "hasWarnings": false,
          "submitted": true,
          "warningsMap": Object {},
        },
        "meta": Object {
          "contacts": Object {
            "email": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "",
              "focused": false,
              "warning": "",
            },
          },
          "profile": Object {
            "firstName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "Field required.",
              "focused": false,
              "warning": "",
            },
            "lastName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "Field required.",
              "focused": false,
              "warning": "",
            },
          },
        },
        "values": Object {
          "contacts": Object {
            "email": undefined,
          },
          "profile": Object {
            "firstName": undefined,
            "lastName": undefined,
          },
        },
      },
      "values": Object {
        "contacts": Object {
          "email": undefined,
        },
        "profile": Object {
          "firstName": undefined,
          "lastName": undefined,
        },
      },
      "wizard": undefined,
    },
  ],
]
`;
